### ISSUE TITLE
Checklist for text page upgrade launch: bilara, front-end, and data

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#1592]
# Note: I am now clearing this checklist and it only contains issues blocking launch

## Overview

For some months the team has been working on an upgrade to the main text pages. This is a complex upgrade which intersects across multiple team domains. In this issue I aim to make a comprehensive checklist for all essential aspects of this upgrade. 

Check off the items once it is live on staging. Once all these items are ticked off, we go live.

This focuses on those things that are essential for **first launch of the new text pages**. A few items, however, may not be required for launch.

Several issues involve both back end and front end, especially those relating to data. We have cleaned up and altered our data formats, and need to make sure the correct data is being applied. This involves: 

- ensuring the correct data is being loaded into the database
- ensuring it is properly exposed in the API
- ensuring that the front end actually uses it

We have put a lot of work into ensuring our data is clean and reliable. **Avoid transforming data on the front end as much as possible**, including things like converting case, etc. If you notice a problem with the data, tell us and we'll fix it.

There is a birds-eye overview of the data flow here:

https://github.com/suttacentral/suttacentral/wiki/Data-flow-for-Bilara-and-SC

**STXNext**: items that may be suitable for STXNext are marked with 🖖. Live long and prosper!

## Final issues to be fixed before launch

- [x] Error when opening more-menu #2048
- [x] Prev/next localization and breadcrumbs are still buggy #2046
- [x] Make sure publication data only appears with the correct text #2042
- [x] Publication data for Bramali Vinaya does not appear. #2041

## Remaining front-end issues: Jakub and Hongda

**high priority bugs**

- [x] Suttaplex name API fixes #2026
- [x] Warning: Content Security Policy of your site blocks the use of 'eval' in JavaScript` #2018
- [x] Chrome on Android freezes when uses text tools #1993
- [x] Review webpack production config #1971
- [x] add publication info to segmented texts #1743 🖖

## Remaining Back end issues: Alex

- [x] Implement pali-hyphenator.js #1690
- [x] Fix swagger at /api/docs #1982
- [x] Use array for segment data #1992 
- [x] Fix bugs in nilakkhana processing of counters #1998

### APIs: high priority, essential: STX Prime Mission back end

- [x] Load `name` and `tree` data to ArangoDB #1806 🖖
- [x] Get blurb info #1778 🖖
- [x] Get publication info #1779 🖖
- [x] Get dictionary info #1780 🖖
- [x] Get navigation data #1781 🖖
- [x] Use new source for glossary #1886 🖖
- [x] Optimize DB with indexing #1947 🖖

### texts: high priority, essential: STX Prime Mission front end

- [x] Top sheet still displays when you navigate away. #2010
- [x] Publication info in top sheet for legacy texts is not showing.
- [x] Navigation: remaining issues #2012
- [x] Migrate to Stripe Checkout for payments, review Stripe security #1614 🖖
- [x] Pages not showing in Firefox & Chrome #2003
- [x] Nilakkhana transformation adds HTML entities to markup #2011
- [x] Breadcrumb nav: URLs and breadcrumbs must be consistent and correct no matter how you get there #1788 
- [x] Context toolbar sometimes shows UID #1994
- [x] Static page Tab bar shows wrong pages on Firefox #2008- [x] toolbar navigation transitions are not great #1995
- [x] Implement reference settings in "views" top sheet. #1754 🖖
- [x] Segmented texts are incomplete #1981
- [x] root and translation are duplicated in vaggasuttas #1955
- [x] Legacy text collections are not showing #1956 #1950
- [x] 404 pages are soft 404 #1152 — still valid?
- [x] Polymer remove: Update package files etc. #1966
- [x] Use LitElement, remove Polymer #1310
- [x] Use new data sources for dictionaries, apply in both search results and lookup. #1508 
- [x] make proper links for Home page sutta quotes #1845 🖖
- [x] make theme changer work on every page #1929 🖖
- [x] Use mwc for search input and fix responsive behavior #1930 🖖
- [x] Redesign routing to improve performance  #1949
- [x] Add field for display of notes in top sheet #1809 🖖
- [x] Use top sheet for publication info #1805 🖖
- [x]  Ensure proper display of root texts from Bilara #1649 🖖
- [x]  Ensure incorrect references do not appear in text #1650 🖖
    - all segmented texts must use bilara-data references only. legacy texts still have inline reference data (eg. Chinese and Sanskrit root texts.)
- [x] Use bottom sheet for lookup tools #1485 
   - mostly complete. blocked by #1508. <s>Also there is a bug where the bottom sheet displays by default.</s>
- [x] Use updated bilara-data for texts and references #1752 🖖
- [x] Refine parallels feature on text pages  #1786 
- [x] Ensure segmented and legacy text pages with subheadings have a Table of Contents #1556 🖖

### texts: low priority, stretch goals

- [x] Ensure script converter works with ṁ #1753 
- [x] Apply `translate='no'` to root texts #1777

### navigation: high priority, essential

- [x] Don't use division folder  #1909
- [x] Remove Collection name or ID from nav card nerdy row #1914
- [x] Implement Gingerbread navigation: #1673 
- [x] Implement new navigation data sources #1714 🖖
- [x] Add blurbs to nav cards #1661 
- [x] Add shortcuts to navigation #1665 
- [x] Nav cards titles are not using correct sources #1660 🖖
    - the sources are as per #1752 
- [x] Likewise for the titles in the context toolbar #1676 🖖
- [x] The title for search results is incorrect. #1676 
- [x] Breadcrumbs should use acronym (`acro`) where possible #1672 
- [x] App bar should disappear and reappear on scroll (except for Home page), as in current SC. #1693
- [x] nav cards steps should have appropriate URL and be strongly tested. #1689 
- [x] fix right/center click in nav cards #1735 
- [x] Go from text pages to the suttaplex for that sutta. #1758
- [x] More-menu is not always on top of everything else #1798 
- [x] Give context toolbar responsive height on text pages #1792 
- [x] Add acronym to navigation-nerdy-row
- [x] Remove popup suttaplex code #1922
- [x] suttaplex share menu has broken #1935

### navigation: low priority, stretch goals

- [x] Yellow brick road should display number of texts at all levels #1663 
- [x] Nav card subtitles should show language where useful #1893 

### general: low priority

- [x] Swath preprocessor for Thai #1103
- [x] Fix font-loading, use only self-hosted fonts #1719 🖖
- [x] Fix video loading; ensure all references to vimeo are removed. #1757 
- [x] Code cleanup #1755 #1925 🖖
- [x] Remove `sutta.json` #1771
- [x] testing and optimizing #1756 🖖
- [x] purge pootle from system #1790 

## bilara: Blake

- [x] Ensure all texts served by Bilara are correct #1588
- [x] Ensure data is getting smoothly from bilara-data to sc-data to arangodb. #1559 🖖
- [x] Ensure the publication model is implemented using "unpublished" and "published" on bilara-data. #1589
- [x] Import Portuguese texts from Pootle https://github.com/suttacentral/bilara/issues/68 🖖
- [x] Add support for SuttaCentral UI translation. https://github.com/suttacentral/bilara/issues/59🖖

## Pali texts and references: STXNext, Robbie, Tracy & Sujato

- [x] Segment remainder of Pali texts. #920 
- [x] Ensure segment numbers use consistent pattern #1591
- [x] Ensure that all instances of the same text use identical segment IDs.
- [x] Double and triple check all text files against accidental loss #1601 
- [x] test and fix doubtful segment IDs. 
   - https://github.com/suttacentral/bilara-data/issues/10, 
   - https://github.com/suttacentral/bilara-data/issues/4
- [x] Add sanitizer test for data (this is an old issue, it may duplicate others)
   - https://github.com/suttacentral/bilara/issues/2
- [x] Add Pali concordance data to bilara/reference #1504 
- [x] Double and triple check bilara/reference for data loss, spelling consistency, etc. See #1473 
- [x] fix cases of incorrect segment assignment marked by `$` in Vinaya. (47 in root, 2 in translation)
- [X] Put /structure/division data into files for tree and names; delete everything else #1714
- [x] check anuvaggas #1773 

## some bits and bobs: Sujato

- [x] Assess and adapt markup, styles, and tooling for paragraph numbers, variants, and references in legacy texts. #1621 
- [x] Some markup files have reference or something in them? an11.992-1151:5.1, an11.22-29:5.1, hmm it looks like all the references for after an11.22 are in the html files.
- [x] fix inconsistent segment ID\d</span>s and markup for debaked suttas
- [x] check span class=speaker
- [x] Ensure bilara texts follow new HTML standards. #1493 
- [x] test all the `_html.json` files by wrapping as HTML and running Tidy.
- [x] Combine texts for bi-pc 91-93
- [x] Test, test, test!
- [x] Check styles for updated text pages.
- [x] Hand adding of a few references from `sc-pts-undefined`, see [here](https://github.com/suttacentral/sc-data/tree/master/misc).
- [x] Resolve discrepancy in sn46.35-38 
   - https://github.com/suttacentral/bilara/issues/40
- [x] Fix "beginning" and "ending" notes #1068 
- [x] fix bad segmenting of … at end of thag/thig
- [x] refactor lists to remove all inline lists. #1678
- [x] Style scroll bars #1582
- [x] Use avif for images
- [x] Ensure 100% compliance with Lighthouse a11y

## HTML clean: Aminah

- [x] Ensure cleaned HTML texts are ready to go, with any 2-dos or explanations that might be needed.
   - https://github.com/suttacentral/suttacentral/wiki/HTML-structure:-template-&-guiding-principles
   - https://github.com/suttacentral/suttacentral/wiki/Legacy-text-coding-guide-(working-draft)

### done!

- [x] Rebuild the text pages on LitElement #1310 
- [x] Serve the texts from bilara with appropriate markup #1482 
  - **Note:** At this time, do not serve `comment` field; the comments are not ready for publication. But we should support all other fields. And we should be ready to serve comments later this year.
- [x] Implement CSS styles #1483 
- [x] Import upgraded legacy html texts from html-clean5 #1491 
- [x] Correct next/prev #1462 
- [x] Fix PWA for Safari, etc. #1708
- [x] Redesign the text settings as "top sheet" #1496 
    - Mostly complete. Outstanding issues above
- [x] Make user settings persist; remove "remember settings" option from top sheet #1748
- [X] Add parallels button to text pages

**Comment by sujato** [comment]
I have removed a few items as non-essential for launch of new text pages:

- items for parallels are included in epic "parallels development overview" https://github.com/suttacentral/suttacentral/issues/1169
- items for importing some remaining references are in bilara repo: https://github.com/suttacentral/bilara/issues/60 https://github.com/suttacentral/bilara/issues/36
